lib/sensor/Sensor.h

```diff
@@ -0,0 +1,27 @@
+#ifndef SENSOR_H
+#define SENSOR_H
+#include "settings.h"
+
+template <typename T>
+class Sensor {
```

Comment:
Might be good to add a virtual destructor (I always add one to top level classes in a hierarchy)

I think doing without is ok if you initialize all your subclass objects statically, but if you want to do something like:

```
Sensor* sensor = new SensorSubclass();

// do some things and pass obj around

delete sensor;
```
The compiler will complain because it won't know how to delete it (ie. won't be able to infer its type and find the right destructor to call)